ee/hogai/tools/session_summarization.py

```diff
@@ -61,6 +66,23 @@ class SessionSummarizationToolArgs(BaseModel):
           * and similar
         """
     )
+    should_use_current_session: bool = Field(
+        description="""
+        - Whether to use the current session ID from user's UI (the session they are viewing) to summarize.
+        - IMPORTANT: Should be always `false` if `current_session_id` is not present in the `search_session_recordings` context.
+        - Examples:
+          * Set to `true` if ALL of the following conditions are met:
+            - `current_session_id` is present in the context
+            - the user wants to summarize a single specific session they are viewing
+            - the user refers to "this session", "the session", "current session", "session I'm looking at", "session I'm watching", or similar singular references
+          * Set to `false` if ANY of the following conditions are met:
+            - `current_session_id` is not present in the context
+            - the user asks to summarize multiple sessions (e.g., "these sessions", "all sessions", "sessions from yesterday")
+            - the user wants to follow current filters
+            - the user specifies search criteria or filters (e.g., "sessions from user X", "mobile sessions")
+            - the user asks to find or search for sessions
+        """,
+    )
```

Comment:
`specific_session_ids_to_summarize` would be a more effective arg. Basically, we should delegate as much choice of concrete values to the top-level agent as we can, to minimize black box behavior.

Example: with `should_use_current_session`, if the user just pastes in a session ID they're interested in, session summaries _won't_ be able to use them. If we make this `specific_session_ids_to_summarize`, then the top-level agent has all the information to either not pass it, or pass based on UI state, or pass based on user input - everything you expect as a user.